### ISSUE TITLE
Build farmer handoff 2017-10-17

### ISSUE THREAD

**Issue by nuclearsandwich** [ros2/build_farmer#61]
#### Tasks in progress

Dosa High Sierra upgrade: When SIP was disabled (requirement for running tests), Dosa was very slow. Dosa currently has SIP enabled and some workarounds to get it to work (which upgrading python will break), so is only being used for CI jobs. It needs to be reinstalled.

#### Tasks pending

- [ ] Update Java on Windows (#43)
- [ ] apt update on Ubuntu hosts (hopefully solving #30)
- [ ] Investigate packet.net host issue (#42)
- [ ] Upgrade eatable to Windows 10

#### Nice-to-haves

- Try VS2017 and assess if we can migrate Windows nodes to it

#### Recurring tasks

- system and package (brew/chocolatey) updates on MacOS and Windows machines
  - [ ] dosa (last updated 2017-09-13)
  - [ ] mini1 (last updated 2017-09-13)
  - [ ] mini2 (last updated 2017-09-13)
  - [ ] portable (last updated 2017-09-13)
  - [ ] windshield (last updated 2017-09-13)

## Test suite health

Currently warnings on release jobs from https://github.com/ros2/rosidl/pull/241

Nightlies on OS X often don't start until the morning because the machines go offline overnight. I've been restarting the repeated job and if it is blocking too many CI jobs or if PRs get merged before it started then I cancel it (so we don't have to wonder if that repeated job was using the same code state as the others).

**Comment by dhood** [comment]
@nuclearsandwich updated. I'll hang onto dosa.

**Comment by dirk-thomas** [comment]
> Try VS2017 and assess if we can migrate Windows nodes to it

This is blocked on having our dependencies like Connext provide VS 2017 packages.

**Comment by nuclearsandwich** [comment]
Closing old handoff issues.
